rust/src/dataset.rs

```diff
@@ -319,14 +407,53 @@ impl Dataset {
     pub fn fragments(&self) -> &Arc<Vec<Fragment>> {
         &self.manifest.fragments
     }
+
+    /// Read all indices of this Dataset version.
+    pub async fn load_indices(&self) -> Result<Vec<Index>> {
```

Comment:
So each scan has to re-read the indices? Would it make sense to just keep it in memory ?

Reply:
That makes sense. Let's keep that as future work? So is caching the IVF partitions, assuming there will be strong locality during queries.